src/qibolab/sweeper.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy.typing as npt
+
+
+@dataclass
+class Sweeper:
+
+    parameter: str
+    values: npt.NDArray
+    pulses: Optional[list] = None
+    qubits: Optional[list] = None
+
+    @property
+    def pulse_type(self):
+        if self.pulses is not None:
+            pulse_type = self.pulses[0].type.name.lower()
+            for pulse in self.pulses:
+                assert pulse.type.name.lower() == pulse_type
+            return pulse_type
```

Comment:
Use the enum:
```suggestion
    def pulse_type(self) -> Optional[PulseType]:
        if self.pulses is not None:
            pulse_type = self.pulses[0].type
            for pulse in self.pulses:
                assert pulse.type is pulse_type
            return pulse_type
```

or even better:
```suggestion
    def pulse_type(self) -> Optional[PulseType]:
        types = set([p.type for p in self.pulses])
        if len(types) > 1:
            raise RuntimeError("Not homogeneous pulses")
        return next(iter(types), None)
```